### PR TITLE
(SIMP-9654) - 389-DS Account Management

### PR DESCRIPTION
**Pull request by trevor-vaughan**
Updated the default instance LDIF template:
  * Allow for account creation
  * Allow for management of accounts with the dsidm system
  * Add the default global password policy
  * Add a custom password policy for 'Host' entries like 'hostAuth'

SIMP-9656 #close
SIMP-9657 #close

### CODE REVIEW

File: spec/acceptance/suites/default/50_sssd_spec.rb [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+require 'spec_helper_acceptance'
+
+describe 'sssd' do
+  hosts.each do |host|
+    let(:hieradata) do
+      {
+        'simp_options::pki'             => true,
+        'simp_options::pki::source'     => '/etc/pki/simp-testing/pki',
+        'simp_options::clamav'          => true,
```
Comment (lnemsick-simp) [inline spec/acceptance/suites/default/50_sssd_spec.rb:11]: Don't think simp_options::clamav is a thing anymore

File: spec/acceptance/suites/default/50_sssd_spec.rb [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        'simp_options::firewall'        => true,
+        'simp_options::haveged'         => true,
+        'simp_options::logrotate'       => true,
+        'simp_options::pam'             => true,
+        'simp_options::selinux'         => true,
```
Comment (lnemsick-simp) [inline spec/acceptance/suites/default/50_sssd_spec.rb:16]: simp_options::selinux no longer exists

**Comment by trevor-vaughan** [comment]
The modifications to the GitHub Actions [allow this to pass](https://github.com/trevor-vaughan/pupmod-simp-ds389/actions/runs/810827605), but the changes won't be picked up until after this PR has been merged.

File: spec/acceptance/suites/default/50_sssd_spec.rb [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        else
+          response = YAML.safe_load(on(host, %(puppet resource service sssd --to_yaml)).stdout.strip)
+          expect(response['service']['sssd']['ensure']).to eq('running')
+          expect(response['service']['sssd']['enable']).to eq('true')
+        end
+      end
+
+      it 'can find testuser' do
+        if host.which('dsidm').empty?
+          skip(dsidm_skip_message)
```
Comment (jeannegreulich) [inline spec/acceptance/suites/default/50_sssd_spec.rb:126]: The metadata.json has EL7 being the only OS supported but you  are skipping testing el7.  SHould the metadata.json be updated?

**Review by jeannegreulich — COMMENTED** [review]
I can't change the password of my user using passwd/  Am I supposed to be able to do this  after sssd is set up?

**Review by jeannegreulich — CHANGES_REQUESTED** [review]
You have the default for $source in instance/tls.pp set incorrectly.  It should be undef to get pki::copys default or /etc/pki/simp/x509.
THe default you have causes an error because it is a relative patch.

**Review by jeannegreulich — APPROVED** [review]

**Pull request merged**